### PR TITLE
Add support for uploading multiple files to server

### PR DESCRIPTION
**Pull request by cypherbeerus**
#899 
Changed file upload object to a list, which allows multiple files to be uploaded at once.

### CODE REVIEW

File: BTCPayServer/Controllers/ServerController.Storage.cs [hunk 1]
```diff
@@ -148,23 +148,42 @@ public enum TmpFileTimeType
...
+                List<string> fileIds = new List<string>();
+                foreach (IFormFile file in files)
                 {
-                    Message = "Invalid file name",
-                    Severity = StatusMessageModel.StatusSeverity.Error
-                });
-                return RedirectToAction(nameof(Files));
+                    if (!file.FileName.IsValidFileName())
+                    {
+                        this.TempData.SetStatusMessageModel(new StatusMessageModel()
```
Comment (Kukks) [inline BTCPayServer/Controllers/ServerController.Storage.cs:160]: You should probably do a counter of invalid file names  and set the status message based on the counter result (if counter ==0 show "Files added", else if counter >0 & counter< files.length show "only x files added due to invalid files" else show "files could not be added"

File: BTCPayServer/Controllers/ServerController.Storage.cs [hunk 2]
```diff
@@ -148,23 +148,42 @@ public enum TmpFileTimeType
...
+                    }
+                    var newFile = await _FileService.AddFile(file, GetUserId());
+                    fileIds.Add(newFile.Id);
+                }
+                if (fileIds.Count == 1)
+                {
+                    return RedirectToAction(nameof(Files), new
+                    {
+                        statusMessage = "File added!",
+                        fileId = fileIds[0]
```
Comment (Kukks) [inline BTCPayServer/Controllers/ServerController.Storage.cs:175]: Perhaps due to this change, we should enhance the `Files` action to accept an array of `fileIds` instead. We can do this in a separate PR.

**Review by Kukks — CHANGES_REQUESTED** [review]
Left a comment around the setting of the status message, and tests need to be fixed.

File: BTCPayServer/Controllers/ServerController.Storage.cs (hunk 2 above)
Comment (cypherbeerus) [reply BTCPayServer/Controllers/ServerController.Storage.cs:175]: Yes, this makes sense. I'll create a separate issue later for this.

**Review by Kukks — CHANGES_REQUESTED** [review]
Looks great now, good job on your first contribution!

**Review by Kukks — APPROVED** [review]
Oops, my evil twin wanted to reject this.  I'll approve :)

**Pull request merged**